src/java/com/sarality/app/view/list/ComponentComparator.java

```diff
@@ -0,0 +1,11 @@
+package com.sarality.app.view.list;
+
+/**
+ * Compares two objects
+ *
+ * @author sunayna@dothat.in sunayna
+ */
+public interface ComponentComparator<T> {
```

Comment:
You are using this as a Filter rather than a Comparator here. What you need is a FilterCriteria or DataFilter that has a single method

passes(T data)

The char sequence is taken in the constructor. We may add another filter tomorrow that filters out Contacts that don't have email addresses. We don't want that boolean parameter to show up in the method signature.